elastic/beats
# Fix: select default agent policy if no enrollment token provided

## What does this PR do?



This PR fixes the non-determinism in agent's policy selection. With this bugfix, the agent will always enroll using the default policy unless the custom enrollment token is selected.

## Why is it important?



Currently the enrollment script (Docker) selects the newest policy. It may happen that the selected policy is not active or it is not intended to be used. 

## Checklist



~- [ ] I have commented my code, particularly in hard-to-understand areas~
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
~- [ ] I have added tests that prove my fix is effective or that my feature works~

## Author's Checklist



## How to test this PR locally



1. Create new custom policy in the Kibana UI.
2. Start new Docker container with Elastic Agent, don't provide the enrollment token.

Expected:
3. Elastic Agent should be started with Default policy, not the newest one (just added).

## Related issues

 
- https://github.com/elastic/beats/issues/23928

## Use cases



## Screenshots



## Logs
